cmd/jujud/agent/machine.go

```diff
@@ -1260,24 +1260,22 @@ func (a *MachineAgent) upgradeWaiterWorker(name string, start func() (worker.Wor
 		logger.Debugf("upgrades done, starting worker %q", name)
 
 		// Upgrades are done, start the worker.
-		worker, err := start()
+		w, err := start()
 		if err != nil {
 			return err
 		}
 		// Wait for worker to finish or for us to be stopped.
-		waitCh := make(chan error)
+		done := make(chan error, 1)
```

Comment:
I'd love too, but it's not a channel :(

On Wed, Jun 15, 2016 at 12:21 PM, Tim Penhey <notifications@github.com>
wrote:

> In cmd/jujud/agent/machine.go
> <https://github.com/juju/juju/pull/5564#discussion_r67090236>:
>
> >  		if err != nil {
> >  			return err
> >  		}
> >  		// Wait for worker to finish or for us to be stopped.
> > -		waitCh := make(chan error)
> > +		done := make(chan error, 1)
>
> Why bother with the don channel at all? Why not just select on w.Wait() ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/juju/juju/pull/5564/files/6440a4de17a3c6ff99a44fab67be7d3b20114cfe#r67090236>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAcA3DDCWJnpjOhaXB9IEDz5XRmhKJXks5qL2GogaJpZM4IwhyC>
> .
>